server/src/main/java/org/elasticsearch/rest/action/admin/cluster/RestCreateSnapshotAction.java

```diff
@@ -54,6 +55,7 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         request.applyContentParser(p -> createSnapshotRequest.source(p.mapOrdered()));
         createSnapshotRequest.masterNodeTimeout(request.paramAsTime("master_timeout", createSnapshotRequest.masterNodeTimeout()));
         createSnapshotRequest.waitForCompletion(request.paramAsBoolean("wait_for_completion", false));
+        createSnapshotRequest.indicesOptions(IndicesOptions.fromRequest(request, createSnapshotRequest.indicesOptions()));
```

Comment:
I am missing why we need to add support for these params as request parameters. Aren't they part of the request body for this API? Are they still supported in the body, meaning do we parse them back when reading the request body?

Reply:
`indicesOptions` is never a part of request body (e.g. [RequestConverters#deleteIndex](https://github.com/elastic/elasticsearch/blob/1d114071dabc8b5208c797a9bbc3b990680577b7/client/rest-high-level/src/main/java/org/elasticsearch/client/RequestConverters.java#L160))  - the biggest question is here - shall we drop in from request itself (i have some doubt on that) and do follow rest api spec - or we have to maintain it properly and add to rest api spec to reflect it ?

Reply:
@vladimirdolzhenko you are referring to delete index, but create snapshot is a different API . See its docs: https://www.elastic.co/guide/en/elasticsearch/reference/6.3/modules-snapshots.html#_snapshot . It seems like ignore_unavailable (which is one of the indices options, as well as index, are specified in the request body. That's why I am asking on what we intend to do here.

Reply:
@javanna I see your point - quite inconsistency here - sorry @jdconrad to mislead you

Reply:
@javanna @vladimirdolzhenko I like the consistency and methods available for switching to indices options the way delete index is doing it; however at the same time, it's probably not worth deprecating the current way right now.  My instinct is to revert this to how it was for now, and we can decide if we want to switch them to parameters at a later time.  Would that work for you guys?

Reply:
yes that's a good idea. let's discuss API changes separately.

Reply:
@javanna Will do.  Thanks for the review here!
@vladimirdolzhenko No need to apologize!  What you showed me makes a lot of sense for a future improvement.

Reply:
Reverted.